From fba9c0961c1701f2a2cd02d9a56923111dc11b09 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 07:08:45 +0200
Subject: [PATCH] avformat/dashdec: Skip LCEVC grouping for representations
 without codecs

The fix is the one recommended in the report.

Fixes: NULL pointer dereference
Fixes: XmvwOQImSoeX
Fixes: AISLE-2026-0111-00259
NULL pointer dereference Replicated through UnModified FFmpeg (ffprobe) with UBSAN
Regression since: e7696357de
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
---
 libavformat/dashdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index f3985788f40f..ef33ff5ff397 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -2287,7 +2287,7 @@ static int dash_read_header(AVFormatContext *s)
     for (i = 0; i < c->n_videos; i++) {
         struct representation *ref;
         rep = c->videos[i];
-        if (!rep->dependencyid || !rep->nb_assoc_stream)
+        if (!rep->dependencyid || !rep->codecs || !rep->nb_assoc_stream)
             continue;
         for (j = 0; j < c->n_videos; j++) {
             if (j == i)
